aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/StorageAdapterInstrumentedTestBase.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.storage.sqlite;
+
+import android.content.Context;
+import android.os.StrictMode;
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.ModelProvider;
+import com.amplifyframework.core.model.ModelSchema;
+import com.amplifyframework.core.model.query.predicate.QueryPredicate;
+import com.amplifyframework.datastore.DataStoreException;
+import com.amplifyframework.datastore.storage.StorageItemChange;
+import com.amplifyframework.testmodels.commentsblog.AmplifyModelProvider;
+import com.amplifyframework.testutils.Await;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test the functionality of {@link SQLiteStorageAdapter} operations.
+ */
+@SuppressWarnings("DesignForExtension") // Utility methods shouldn't be overwritten
+public abstract class StorageAdapterInstrumentedTestBase {
```

Comment:
If they are utility methods, then I recommend to put them into utility classes, not into an abstract base.

Abstract classes are good candidates for enforcing policy on an interface. But, that’s not really what’s needed here. Here, you’re just trying to re-use some utility code in a few places.

Reply:
The biggest reason for using abstract class (it can just be a non-final class to be honest) here was to allow re-initialization of `SqliteStorageAdapter` per test case without having to copy and paste the same code into each individual test. The "utility methods" make a direct reference to this instance, so I can't move them to a static utility class either.